sklearn/decomposition/fastica_.py

```diff
@@ -563,8 +565,7 @@ def inverse_transform(self, X, copy=True):
         """
         check_is_fitted(self, 'mixing_')
 
-        if copy:
```

Comment:
Yes `X += self.mean_` made numpy 0.10 raise an error since X is not explicitly float but `mean_` was... (is this fix okay or should I rather do `check_array(X, dtype=FLOAT_DTYPES)`?)